frontend-mobile/lib/widgets/coffee_call.dart

```diff
@@ -1,55 +1,62 @@
 import 'package:coffee_caller/communication/coffee_caller_protocol.dart';
 import 'package:coffee_caller/communication/socket_client.dart';
-import 'package:coffee_caller/view/cubit/coffee_call_cubit.dart';
-import 'package:coffee_caller/view/cubit/coffee_call_state.dart';
-import 'package:coffee_caller/view/settings_screen.dart';
+import 'package:coffee_caller/storage/settings_storage.dart';
+import 'package:coffee_caller/widgets/no_username.dart';
+import 'package:coffee_caller/widgets/cubit/coffee_call_cubit.dart';
+import 'package:coffee_caller/widgets/cubit/coffee_call_state.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
-class CoffeeCallScreen extends StatelessWidget {
-  const CoffeeCallScreen({
+class CoffeeCall extends StatelessWidget {
+  const CoffeeCall({
     super.key,
   });
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        title: const Text('Coffee Caller'),
-        actions: [
-          TextButton.icon(
-            style: TextButton.styleFrom(foregroundColor: Colors.white),
-            onPressed: () {
-              Navigator.push(
-                context,
-                MaterialPageRoute(builder: (context) => const SettingsScreen()),
-              );
-            },
-            icon: const Icon(Icons.settings),
-            label: const Text('Settings'),
-          ),
-        ],
-      ),
-      body: BlocProvider(
-        create: (_) => CoffeeCallCubit(
-          socket: context.read<SocketClient>(),
-          protocol: CoffeeCallerProtocol(
-            socketClient: context.read<SocketClient>(),
-          ),
-        )
-          ..init()
-          ..connect(),
-        child: const CoffeeCallBody(),
-      ),
+    return BlocProvider(
+      create: (_) => CoffeeCallCubit(
+        socket: context.read<SocketClient>(),
+        protocol: CoffeeCallerProtocol(
+          socketClient: context.read<SocketClient>(),
+        ),
+      )
+        ..init()
+        ..connect(),
+      child: const CoffeeCallBody(),
     );
   }
 }
 
-class CoffeeCallBody extends StatelessWidget {
+class CoffeeCallBody extends StatefulWidget {
   const CoffeeCallBody({super.key});
 
+  @override
+  State<StatefulWidget> createState() => _CoffeeCallBodyState();
+}
+
+class _CoffeeCallBodyState extends State<CoffeeCallBody> {
+  late String userName = '';
```

Comment:
Once again this feels a little bit weird because initState cannot be async

Reply:
That's totally fine. If you want to we can refactor this together to use a cubit, then you'll see how these work, but not sure if that's worth the effort.